db/dcrpg/queries.go

```diff
@@ -3619,6 +3620,182 @@ func appendBlockFees(charts *cache.ChartData, rows *sql.Rows) error {
 	return rows.Err()
 }
 
+// retrievePrivacyParticipation retrieves the sum of all mixed vouts that is newer than the data
+// in the provided ChartData. This data is used to plot fees on the /charts page.
+// This is the Fetcher half of a pair that make up a cache.ChartUpdater.
+func retrievePrivacyParticipation(ctx context.Context, db *sql.DB, charts *cache.ChartData) (*sql.Rows, error) {
+	rows, err := db.QueryContext(ctx, internal.SelectMixedTotalPerBlock, charts.TotalMixedTip())
+	if err != nil {
+		return nil, err
+	}
+	return rows, nil
+}
+
+// Append the result from retrievePrivacyParticipation to the provided ChartData. This
+// is the Appender half of a pair that make up a cache.ChartUpdater.
+func appendPrivacyParticipation(charts *cache.ChartData, rows *sql.Rows) error {
+	defer rows.Close()
+	blocks := charts.Blocks
+	for rows.Next() {
+		var blockHeight uint64
+		var totalMixed int64
+		if err := rows.Scan(&blockHeight, &totalMixed); err != nil {
+			log.Errorf("Unable to scan for MixedVoutsPerBlock fields: %v", err)
+			return err
+		}
+		if totalMixed < 0 {
+			totalMixed *= -1
+		}
```

Comment:
In what circumstances does this happen?  Surely that indicates a bug in something else.

Reply:
Please comment on this case.

Reply:
I no longer have this issue, this was based on different implementation where my effort to remove `spent vouts` was leading the negative value. I ought to have removed it when the implementation was changed.